File: src/main/java/jb/integrator/Integrator.java
package jb.integrator;

import de.bwaldvogel.liblinear.Model;
import jb.config.Opts;
import jb.data.Dataset;
import jb.data.IntegratedModel;
import jb.data.SelectedTuple;

import java.util.List;

@FunctionalInterface
public interface Integrator {

    IntegratedModel integrate(SelectedTuple selectedTuple, List<Model> clfs, Dataset dataset, Opts opts);

}

File: src/main/kotlin/jb/integrator/Integrator.kt
package jb.integrator

import de.bwaldvogel.liblinear.Model
import jb.config.Opts
import jb.data.Dataset
import jb.data.IntegratedModel
import jb.data.SelectedTuple

interface Integrator {

    fun integrate(selectedTuple: SelectedTuple, clfs: List<Model>, dataset: Dataset, opts: Opts): IntegratedModel

}
